src/NuGet.Core/NuGet.Commands/Utility/MSBuildProjectFrameworkUtility.cs

```diff
@@ -221,16 +298,37 @@ public static NuGetFramework GetProjectFrameworkReplacement(NuGetFramework frame
             return framework;
         }
 
-        private static string GetPropertyOrNull(string propertyName, IDictionary<string, string> projectProperties)
+        /// <summary>
+        /// Get a NuGetFramework out of the passed object. The argument is expected to either be a <see cref="NuGetFramework"/> or <see cref="string"/>.
+        /// </summary>
+        private static NuGetFramework GetAsNuGetFramework(object arg)
```

Comment:
This makes me kind of uncomfortable, but I'm trying to figure out if there's a good way around it. :\

Reply:
I wanted to avoid NuGetFramework -> ToShortFolderName -> NuGetFramework.Parse fiasco. 

Allowed me to have 1 implementation of everything.

Reply:
How often is `GetProjectFrameworkStrings` called? Is it on hot-paths?  In other words, can we live with the perf-hit?  My biggest gripe is something Kat mentioned in the other comment, I also dislike that this is bypassing built-in typechecking and using typecasting. I can't think of anything better that will definitely will work. I'm getting ideas that I'll try out and I'll post back if it works, but I'm not confident the compiler will accept what I'm thinking of.

Getting even more out-of-scope, is there any good reason for `GetProjectFrameworkStrings` to exist? If the usages just end up calling `NuGetFramework.Parse`, then they should be changed to use your new method and mark the old one obsolete.

Reply:
> Getting even more out-of-scope, is there any good reason for GetProjectFrameworkStrings to exist?

Maybe, I can clean them up in a follow up issue.
I didn't want to try to fix it because there's many usages. 

Regarding when it's being called, it's called on the UI thread in many cases, because that's where we read the props.

Finally `GetProjectFrameworkStrings` is part of the public API, are you ok with me removing it? :)

Reply:
> Finally `GetProjectFrameworkStrings` is part of the public API, are you ok with me removing it? :)

Change the implementation to call the new method and use `.ToString()`, or `.DotNetFrameworkName` or whatever (so it's "slow"), and mark it as obsolete. API compatible, but unused so we don't care about performance.

Reply:
Created https://github.com/NuGet/Home/issues/9878 and put it in S175.